### PR TITLE
Removes v3 and unused v1 API routes

### LINKED ISSUE #1941: Remove old provider specific routes
**User Story**

After https://app.zenhub.com/workspace/o/kubermatic/kubermatic/issues/1940 is done in the frontend we need to remove old routes and the redirect.

Remove this routes from the backend code:
Please map the routes:
* /api/v1/azure/sizes 
* /api/v1/digitalocean/sizes
* /api/v1/digitalocean/sizes
* /api/v1/openstack/securitygroups
* /api/v1/openstack/sizes
* /api/v1/openstack/subnets
* /api/v1/openstack/tenants 
* /api/v1/vsphere/networks 

**Acceptance criteria**

Only new /api/v1/providers/* routes are used

### PR DESCRIPTION
**Pull request by p0lyn0mial**
**What this PR does / why we need it**: simply removes all `v3` and unused `v1` HTTP endpoints

the following unused `v1`  endpoints were removed :
/ssh-keys
/digitalocean/sizes
/azure/sizes
/openstack/*
/vsphere/networks

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #1941 #2023 

**Release note**:

```release-note
API endpoints for clusters not separated into projects have been deprecated.
```

### CODE REVIEW

**Comment by p0lyn0mial** [comment]
do you know any tool that spits out unused code ?

**Comment by kdomanski** [comment]
@p0lyn0mial the golang linter `deadcode` should detect all **unexported** unused symbols.

**Comment by p0lyn0mial** [comment]
> @p0lyn0mial the golang linter deadcode should detect all unexported unused symbols

@kdomanski do you know how does it differ from `unused` linter ?  https://github.com/dominikh/go-tools/tree/master/cmd/unused

**Comment by maciaszczykm** [comment]
@p0lyn0mial GoLand can do it among other nice things, it is not free though.

**Comment by kdomanski** [comment]
@p0lyn0mial ¯\\\_(ツ)_/¯

**Comment by p0lyn0mial** [comment]
trying both `deadcode` and `GoLand`, many thanks :)

File: api/pkg/handler/node_v1.go [hunk 1]
```diff
@@ -314,6 +330,201 @@ func convertNodesV2ToNodesV1(nodesV2 []*apiv2.Node) []*apiv1.Node {
...
+	}
+	for _, machine := range machines {
+		if ref.UID == machine.UID {
+			return &machine
+		}
+	}
+	return nil
+}
+
+func tryToFindMachineAndNode(name string, machineClient clusterv1alpha1clientset.Interface, kubeClient kubernetes.Interface) (*clusterv1alpha1.Machine, *corev1.Node, error) {
```
Comment (maciaszczykm) [inline api/pkg/handler/node_v1.go]: s/tryToFindMachineAndNode/findMachineAndNode?

File: api/pkg/handler/node_v1.go [hunk 2]
```diff
@@ -314,6 +330,201 @@ func convertNodesV2ToNodesV1(nodesV2 []*apiv2.Node) []*apiv1.Node {
 	return nodesV1
 }
 
+func outputNode(node *corev1.Node, hideInitialNodeConditions bool) *apiv2.Node {
```
Comment (p0lyn0mial) [inline api/pkg/handler/node_v1.go:333]: this method and the others were moved from `node_v3.go`

**Comment by p0lyn0mial** [comment]
@kdomanski if you don't mind I'll assign you as a reviewer since this pull is quite heavy.

File: api/pkg/handler/node_v1.go [hunk 3]
```diff
@@ -314,6 +330,201 @@ func convertNodesV2ToNodesV1(nodesV2 []*apiv2.Node) []*apiv1.Node {
...
+		}
+	}
+
+	if node == nil && machine == nil {
+		return nil, nil, nil
+	}
+
+	if node != nil && machine != nil {
+		return machine, node, nil
+	}
```
Comment (maciaszczykm) [inline api/pkg/handler/node_v1.go]: I think ifs from 508 and 512 can be removed without any impact.

**Review by maciaszczykm — COMMENTED** [review]
Looks good to me. I have only 2 minor comments.

**Comment by p0lyn0mial** [comment]
comments addressed, PTAL.

**Review by maciaszczykm — APPROVED** [review]

**Review by kdomanski — APPROVED** [review]
:+1: solid, thorough work

**Pull request merged**